docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/grid-pro_02_datasources.mdx

```diff
@@ -0,0 +1,197 @@
+- Suitable for datasets that are small to medium in size and can be entirely loaded into the client.
+
+### Usage Example
+
+```typescript
+@customElement({
+  name: 'connected-grid-client-side-example',
+  template: html`
+    <rapid-grid-pro enable-row-flashing enable-cell-flashing>
+      <grid-pro-client-side-datasource
+        resource-name="ALL_TRADES"
+        criteria="INSTRUMENT_ID == 'TSLA'"
+      ></grid-pro-client-side-datasource>
+    </rapid-grid-pro>
+  `,
+})
+export class ConnectedGridClientSideExample extends GenesisElement {}
+```
+
+## `<grid-pro-server-side-datasource>`
+
+### Server-Side Features
+
+- **Sorting**: Order by any INDEX column/field, ASC or DESC.
+  - This is a server limitation. For each column that needs sorting, an INDEX must be defined. See more about indexes [here](https://learn.genesis.global/docs/database/data-types/index-entities/).
+  - By default, sorting is disabled on all columns, but if valid indexes are detected from the metadata, those columns/fields will have sorting enabled.
+  - If the user attempts to force `sortable`, we'll check its validity, warn in the logs if not, and mention other available indexes if any.
+- **Filtering**:
+  - Filtering options are automatically generated based on the resource’s metadata.
+  - **String Filters**:
+    - blank
+    - contains
+    - equals
+    - notBlank
+    - notEqual
+    - wordStartsWith
+  - **Number Filters**:
+    - equals
+    - notEqual
+    - greaterThan
+    - greaterThanOrEqual
+    - lessThan
+    - lessThanOrEqual
+    - inRange
+    - blank
+    - notBlank
+  - **Date Filters**:
+    - equals
+    - lessThan
+    - greaterThan
+    - inRange
+    - isToday
+    - blank
+    - notBlank
+
+### Server-Side Notes
+
+- Operations like sorting and filtering are pushed to the backend.
+  - Filtering uses `CRITERIA_MATCH` param.
+  - Sorting uses `ORDER_BY` param.
+  - Once filtering/sorting is applied, the datasource component resets itself, starting a new stream with the updated params.
+- **Limitations**:
+  - Sorting can only be applied to index fields/columns.
+  - `ROWS_COUNT` doesn’t reflect the correct amount when a `CRITERIA_MATCH` (filtering) is applied. For example, if a resource has 100 records and a criteria returns only 50, the server still reports 100. We have to calculate this manually.
+- Suitable for large datasets where only a subset of data is loaded into the client based on user interactions.
+
+### Usage Example
+
+```typescript
+@customElement({
+  name: 'connected-grid-server-side-example',
+  template: html`
+    <rapid-grid-pro enable-row-flashing enable-cell-flashing>
+      <grid-pro-server-side-datasource
+        resource-name="ALL_TRADES"
+        criteria="INSTRUMENT_ID == 'TSLA'"
+      ></grid-pro-server-side-datasource>
+    </rapid-grid-pro>
+  `,
+})
+export class ConnectedGridServerSideExample extends GenesisElement {}
+```
+
+## Common Attributes
```

Comment:
These attribute tables are good but they don't follow the format of every other [API section ](https://github.com/genesiscommunitysuccess/docs/blob/a5615064a003f992eadbd2ae3c960ad766fa5de8/docs/001_develop/03_client-capabilities/008_interaction/anchored-region.mdx?plain=1#L133)

I know we're short on time now so happy for this to go in if you think you can't do it, and then you or I can tweak it when we get the chance